transmart-rest-api/grails-app/controllers/org/transmartproject/rest/QueryController.groovy

```diff
@@ -128,6 +129,59 @@ class QueryController extends AbstractQueryController {
                 })
     }
 
+    /**
+     * Data table endpoint:
+     * <code>/v2/observations/table?type=${type}&constraint=${constraint}&rowDimensions=${rowDimensions}&
+     * columnDimensions=${columnDimensions}&rowSort=${rowSort}&columnSort=${columnSort}&limit={limit}&offset={offset}</code>
+     *
+     * Expects a {@link Constraint} parameter <code>constraint</code>.
+     *
+     * The type should be the data type name of a high dimension type, or 'autodetect'.
+     * Only 'clinical' type is currently supported.
+     *
+     * Expects columnDimensions and rowDimensions parameters.
+     *
+     * Optional rowSort and columnSort parameters allow to define the sorting.
+     *
+     * Pagination is supported via limit and offset parameters.
+     *
+     * @return a tabular representation of hypercube in a json format.
+     */
+    def table() {
+        def args = getGetOrPostParams()
+        checkForUnsupportedParams(args, ['type', 'constraint', 'rowDimensions', 'columnDimensions',
+                                         'rowSort', 'columnSort', 'limit', 'offset'])
+
+        if (args.type != 'clinical') { throw new OperationNotImplementedException("High dimensional data is not yet " +
+                "implemented for the data table")
+        }
+
+        Constraint constraint = bindConstraint((String) args.constraint)
+        User user = (User) usersResource.getUserFromUsername(currentUser.username)
+
+        if (args.limit == null) {
+            throw new InvalidArgumentsException("Parameter 'limit' is required")
+        }
+        int limit = Integer.parseInt((String) args.limit)
+        Long offset = args.offset ? Long.parseLong((String) args.offset) : 0
+
+        def rowSort = parseIfJson(args.rowSort)
+        def columnSort = parseIfJson(args.columnSort)
+        def rowDimensions = parseIfJson(args.rowDimensions)
+        def columnDimensions = parseIfJson(args.columnDimensions)
+
+        [rowDimensions: rowDimensions, columnDimensions: columnDimensions].each { name, list ->
+            if(! list instanceof List || list.any { ! it instanceof String }) {
+                throw new InvalidArgumentsException("$name must be a JSON array of strings")
+            }
+        }
+
+        OutputStream out = getLazyOutputStream(Format.JSON)
+
+        hypercubeDataSerializationService.writeTable(constraint, rowDimensions, columnDimensions,
+                rowSort, columnSort, limit, offset, user, out)
```

Comment:
Or combine some of these parameters in one object (e.g, of a class PageParameters with fields `rowSort`, `columnSort`, `limit` and `offset`).